src/entities/nestedJoin/parseNestedJoinCall.ts

```diff
@@ -0,0 +1,82 @@
+import { Hex } from '../../types';
+import { ZERO_ADDRESS, getPoolAddress } from '../../utils';
+import { WeightedEncoder } from '../encoders';
+import { ComposableStableEncoder } from '../encoders/composableStable';
+import { NestedJoinCall } from './types';
+import { Relayer } from '../relayer';
+import { replaceWrapped } from '../utils/replaceWrapped';
+
+export const parseNestedJoinCall = ({
+    useNativeAssetAsWrappedAmountIn,
+    chainId,
+    sortedTokens,
+    poolId,
+    poolType,
+    kind,
+    sender,
+    recipient,
+    maxAmountsIn,
+    minBptOut,
+    fromInternalBalance,
+    outputReferenceKey,
+}: NestedJoinCall) => {
+    // replace wrapped token with native asset if needed
+    let tokensIn = [...sortedTokens];
+    let value = 0n;
+    if (chainId && useNativeAssetAsWrappedAmountIn) {
+        tokensIn = replaceWrapped([...sortedTokens], chainId);
+        const nativeAssetIndex = tokensIn.findIndex((t) =>
+            t.isSameAddress(ZERO_ADDRESS),
+        );
+        if (nativeAssetIndex > -1) {
+            value = maxAmountsIn[nativeAssetIndex].amount;
+        }
+    }
+
+    const poolAddress = getPoolAddress(poolId);
```

Comment:
Since we're already getting poolAddress in `getNestedJoinCalls` should we use that as another param?